src/compile.rs

```diff
@@ -329,6 +347,158 @@ pub fn build_func_map<'a>(
+                                            &format!("Found duplicate genfscon rules for filesystem {} with differing contexts: {}", rule.fs_name, rule.context)));
+                                        // We just made the error we are okay to unwrap it
+                                        error = Some(error.unwrap().add_additional_message(&inner_rule.file,
+                                            inner_range,
+                                            &format!("Found duplicate genfscon rules for filesystem {} with differing contexts: {}", inner_rule.fs_name, inner_rule.context)));
+                                    }
+                                // Our paths are the same but our file types differ, we must also have a file type.
+                                } else if path == inner_path
+                                    && rule.file_type != inner_rule.file_type
+                                    && rule.file_type.is_some()
+                                {
+                                    let mut inner_range: Option<Range<usize>> = None;
+                                    // We need to look through the function's args to get the range on the file type arg we are looking for
+                                    for arg in &inner_rule.func_call.args {
+                                        if let Argument::List(cas_args) = &arg.0 {
+                                            for cas_arg in cas_args {
+                                                // unwrap is safe here due to else if above
+                                                if *cas_arg
+                                                    == inner_rule.file_type.unwrap().to_string()
+                                                {
+                                                    inner_range = cas_arg.get_range();
+                                                }
+                                            }
+                                        }
+                                    }
+                                    // error is not None so we have already found something, so we just
+                                    // need to add a new error message
+                                    if let Some(unwrapped_error) = error {
+                                        error = Some(unwrapped_error.add_additional_message(&inner_rule.file,
+                                            inner_range,
+                                            &format!("Found duplicate genfscon rules for filesystem {} with differing file types", inner_rule.fs_name)));
+                                    } else {
+                                        // Just like above we need to look through the outer function's args to get the correct range
+                                        let mut outer_range: Option<Range<usize>> = None;
+                                        for arg in &rule.func_call.args {
```

Comment:
This for loop can be its own function that takes the range and the rule, right?

Reply:
Actually, if we store the range in FileSystemContextType, we don't need this loop at all I think.